scipy/stats/stats.py

```diff
@@ -4773,15 +4769,12 @@ def ks_2samp(data1, data2):
     n1 = data1.shape[0]
     n2 = data2.shape[0]
     data_all = np.concatenate([data1, data2])
-    cdf1 = np.searchsorted(data1, data_all, side='right') / (1.0*n1)
-    cdf2 = np.searchsorted(data2, data_all, side='right') / (1.0*n2)
+    cdf1 = np.searchsorted(data1, data_all, side='right') / n1
+    cdf2 = np.searchsorted(data2, data_all, side='right') / n2
     d = np.max(np.absolute(cdf1 - cdf2))
     # Note: d absolute not signed distance
-    en = np.sqrt(n1 * n2 / float(n1 + n2))
-    try:
```

Comment:
There may be an issue with `distributions.kstwobign.sf`.  It calls `special.kolmogorov(y)`, and for small values of `y`, divide-by-zero can occur inside the "C" function `kolmogorov(y)`. 
[ The probabilities returned can also be greater than 1, and `kolmogorov` may take a very noticeable amount to time to compute the probability, but I don't know that either of those would be the cause of an Exception.]
gh-8737 fixes those 3 issues and more.

Reply:
@akhmerov it would be good to remove that commit, and instead add a comment explaining what @pvanmulbregt said may be the reason.

Reply:
I'm still hesitant to insert a blank `except` clause, so I decided to investigate a bit more. I have checked that small inputs to `special.kolmogorov` indeed cause a slowdown. That, or the precision loss are not caught by the try-exception, as expected by @pvanmulbregt. The potential source of division by zero error is [over here](https://github.com/pvanmulbregt/scipy/blob/master/scipy/special/cephes/kolmogorov.c#L95).

There `p` in the denominator is a partial sum of the form
![image](https://user-images.githubusercontent.com/2069677/41200723-683fef30-6caa-11e8-8b1b-deadcf9428c7.png)

For very small `y`, this sum will evaluate to 0 for some even number of terms, however trying to execute `special.kolmogorov(1.2e-16)` does not raise for me in scipy 1.1.0 on ubuntu + conda-forge.

As far as ways to proceed go:

* Restoring the try-except seems to be the worst option to me. It does not resolve the long evaluation time problem or occurrence of probabilities larger than one, and will probably accumulate dust for another decade.
* We could hotpatch the potential division by zero by rewriting `(t / p) > 1.1e-16` into `t > 1.1e-16 * p`. This would be actually *more* correct because it would take care of cases where `p < 0` because of weird round-off errors.
* Finally we could merge this as is, either waiting for #8737, or even without waiting, given that the benefit of try-except is already ephemeral.

@rgommers @pvanmulbregt what do you think?

Reply:
`t > 1.1e-16 * p` sounds fine. Any other improvements to reduce the sensitivity in the tails are also fine to add.

My thinking on this PR is that I'd like to merge it, since it's reviewed and pretty much good to go (while gh-8737 could still take a while). And I don't want any regressions for current users, so there should not be any new exceptions bubbling up. So at least `except <relevant errors>` would be good.

Reply:
That'd be quite fine (that was my motivation to investigate), but I wasn't able to observe the error, and I'm unsure how the C-generated errors would manifest. What if I implement the suggested rewrite of the division instead? That should avoid the zero division error (assuming that is what we're catching), and would avoid the generic `except`.

Because this bit isn't covered with tests, it's very hard for me to confirm that there is an actual regression for the current users.

Reply:
>  it's very hard for me to confirm that there is an actual regression for the current users.

indeed, and in case of doubt it's better to be safe than sorry:) bare `except`s are not inherently bad, and I'm sure it was put in because someone encountered an exception. So if we want to find out if there are exceptions still, you can do something like:

    try:
        ...
    except:
        warnings.warn('This should not happen! Please open an issue at '
                      'https://github.com/scipy/scipy/issues and provide the code '
                      'you used to trigger this warning')

Reply:
> bare excepts are not inherently bad

A bit of an aside here: from what I understand, in most cases if you are trying to be general then you should `except Exception:` so that you do not also catch things like `SystemExit` and `KeyboardInterrupt` which would be caught by a bare `except:`. This seems like one of the cases where `except Exception:` would be more appropriate than `except:`.

Reply:
pushed a commit to revert the try-except removal. time to get this in